src/FieldEncryptProcessEntities.php

```diff
@@ -84,17 +42,37 @@ class FieldEncryptProcessEntities {
   protected $entityManager;
 
   /**
-   * @param \Drupal\encrypt\EncryptService $encrypt_service
+   *
+   */
+  protected $fieldEncryptMap;
+
+  /**
+   * @param \Drupal\field_encrypt\FieldEncryptMapPluginManager $field_encrypt_map_plugin_manager
    * @param \Drupal\Core\Entity\Query\QueryFactory $query_factory
    * @param \Drupal\Core\Entity\EntityManager $entity_manager
    */
-  public function __construct(EncryptService $encrypt_service, QueryFactory $query_factory, EntityManager $entity_manager) {
-    $this->encryptService = $encrypt_service;
+  public function __construct(FieldEncryptMapPluginManager $field_encrypt_map_plugin_manager, QueryFactory $query_factory, EntityManager $entity_manager) {
```

Comment:
This is where you'll want the container.

You will want to implement a static "create" function that returns the parameter of the constructor.

Seen here:

https://github.com/d8-contrib-modules/taxonomy_menu/blob/master/src/Plugin/Derivative/TaxonomyMenuMenuLink.php